sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -0,0 +1,434 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A runner that allows running of Beam pipelines interactively.
+
+This module is experimental. No backwards-compatibility guarantees.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import copy
+import glob
+import os
+import shutil
+import tempfile
+import urllib
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam import runners
+from apache_beam.io import filesystems
+from apache_beam.portability.api import beam_runner_api_pb2
+from apache_beam.runners import pipeline_context
+from apache_beam.runners.interactive import display_manager
+from apache_beam.transforms import combiners
+
+SAMPLE_SIZE = 8
+
+
+class InteractiveRunner(runners.PipelineRunner):
+  """An interactive runner for Beam Python pipelines.
+
+  Allows interactively building and running Beam Python pipelines.
+  """
+
+  def __init__(self, underlying_runner=None):
+    self._underlying_runner = underlying_runner or runners.DirectRunner()
+    self._cache_manager = CacheManager()
+
+  def cleanup(self):
+    self._cache_manager.cleanup()
+
+  def apply(self, transform, pvalueish):
+    # TODO(qinyeli): Remove runner interception of apply.
+    return self._underlying_runner.apply(transform, pvalueish)
+
+  def run_pipeline(self, pipeline):
+    if not hasattr(self, '_desired_cache_labels'):
+      self._desired_cache_labels = set()
+    print('Running...')
+
+    # pylint: disable=import-error
+    from apache_beam.pipeline import Pipeline
+
+    # When possible, invoke a round trip through the runner API.
```

Comment:
Is this still need (or desirable)?

Reply:
You are right.. This is not desirable. Thanks!